src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -1204,26 +1210,40 @@ public void UpdateBubbleContent()
                 ? InfoBubbleViewModel.Style.Error
                 : InfoBubbleViewModel.Style.Warning;
 
-            // NOTE!: If tooltip is not cached here, it will be cleared once the dispatcher is invoked below
-            string content = NodeModel.ToolTipText;
-            if (string.IsNullOrWhiteSpace(content)) return;
             const InfoBubbleViewModel.Direction connectingDirection = InfoBubbleViewModel.Direction.Bottom;
-            var data = new InfoBubbleDataPacket(style, topLeft, botRight, content, connectingDirection);
-
-            ErrorBubble.UpdateContentCommand.Execute(data);
+            var warningTexts = content.Split(new string[] { Environment.NewLine }, StringSplitOptions.None);
+            var packets = new List<InfoBubbleDataPacket>(warningTexts.Length);
+            foreach (var text in warningTexts)
+            {
+                var data = new InfoBubbleDataPacket(style, topLeft, botRight, text, connectingDirection);
```

Comment:
so the meat of this change is that the warning text is now separated by newline characters, and we add individual `InfoBubbleDataPacket`'s for each one?
I'm still a bit confused how this fixes the code block issue.

I also am curious if it affects other features that might rely on the warning being a single unseparated string, or might separate it later? For example the documentation browser features that search for help urls in the warnings?

Reply:
This solution actually doesn't work in all cases, as there could be some warning messages that contain newline characters so I'm looking to see if I could split warnings (InfoBubbleDataPackets) based on the href link, assuming there is a separate link for each warning. @mjkkirschner can you confirm if that's the case?

Also, can you think of a case where a non-CBN node can have more than one warning/error so that I can test it? I'm assuming that that will be a requirement going forwards with the info state thing.

Reply:
* only some warnings have href links, those that do, only have 1.
* a ZT node could have multiple errors potentially in the following ways (unconfirmed)
    * replication yielding different errors for different inputs.
    * Use of a method like https://github.com/DynamoDS/Dynamo/blob/master/src/NodeServices/MessageEvents.cs#L20 `OnLogWarningMessage` to raise a warning, and also a real warning encountered during execution of the node.

Reply:
Thanks. In my experience, I've never seen the replication yielding multiple errors show as multiple messages on a node. Can you reproduce that?

Reply:
In any case, I'm still thinking of a better solution for this.

Reply:
I cannot reproduce it, I may be remembering behavior in 1.x or even earlier. I just tried back in 2.6 and cannot reproduce there, so let's chalk it up to remembering incorrectly.

Though... should that be the behavior? It does not seem correct.  Even if the evaluations yield different errors we only return the first.